README.md

```diff
@@ -74,22 +63,519 @@ include '::auditd'
+  * Default: 5
+
+##### `disp_qos`:
+
+  * Valid Types: 'lossy','lossless'
+  * Default: 'lossy'
+
+##### `dispatcher`:
+
+  * Valid Options: Stdlib::Absolutepath
+  * Default: '/sbin/audispid'
+
+##### `name_format`:
+
+  * Valid Options: Auditd::NameFormat
+  * Default: 'User'
+
+##### `lname`:
+
+  This parameter is an alias for the ``name`` variable in the man pages. Because $name is reserved in the Puppet language, $lname is used to represent the ``name`` key in auditd.conf.
+  * Valid Options: String
+  * Default: $facts['fqdn']
+
+##### `max_log_file_action`:
+
+  * Valid Options: Auditd::DiskFullAction
+  * Default: 'ROTATE'
+
+##### `max_log_file`:
+
+  * Valid Options: Integer[0]
+  * Default: 24
+
+##### `space_left`:
+
+  * Valid Options: Integer[0]
+  * Default: 75
+
+##### `space_left_action`:
+
+  * Valid Options: Auditd::SpaceLeftAction
+  * Default: 'SYSLOG'
+
+##### `action_mail_acct`:
+
+  * Valid Options: String
+  * Default: 'root'
+
+##### `admin_space_left`:
+
+  * Valid Options: Integer
+  * Default: 50
+
+##### `admin_space_left_action`:
+
+  * Valid Options: Auditd::SpaceLeftAction
+  * Default: 'SUSPEND'
+
+##### `disk_full_action`:
+
+  * Valid Options: Auditd::DiskFullAction
+  * Default: 'SUSPEND'
+
+##### `disk_error_action`:
+
+  * Valid Options: Auditd::DiskErrorAction
+  * Default: 'SUSPEND'
+
+##### `default_audit_profile`:
+
+  * Valid Types: 'simp', Boolean]
+  * Default: 'simp'
+
+### auditd::config::audisp
+
+All parameters in this class are documented on page 5 of audispd.conf's man page. The execption is the $specific_name parameter, which maps to the audispd.conf `name` variable.
+
+##### `q_depth`:
+
+  * Valid Options: Integer
+  * Default: 160
+
+##### `overflow_action`:
+
+  * Valid Options: Auditd::OverflowAction
+  * Default: 'SYSLOG'
+
+##### `priority_boost`:
+
+  * Valid Options: Integer
+  * Default: 4
+
+##### `max_restarts`:
+
+  * Valid Options: Integer
+  * Default: 10
+
+##### `name_format`:
+
+  * Valid Options: Auditd::NameFormat
+  * Default: 'USER'
+
+##### `specific_name`:
+
+  maps to `name` in audispd.conf, because $name is restricted in Puppet.
+  * Valid Options: String
+  * Default: $facts['fqdn']
+
+### auditd::config::grub
+
+The `auditd::config::grub` class controls if auditing is enabled at boot time.
+
+##### `enable`:
+
+  If true, audit at boot is enabled.
+  * Valid Options: Boolean
+  * Default: true
+
+### auditd::config::audisp::syslog
+
+The `auditd::config::audisp::syslog` class utilizes rsyslog to send audit records to a remote logging server. **This class REQUIRES the `pupmod-simp-rsyslog` module to be installed**. It is suggested that you read the headers in the manifest for this class if you intend to leverage this functionality.
+
+##### `drop_audit_logs`:
+
+  * Valid Options: Boolean
+  * Default: true
+
+##### `priority`:
+
+  * Valid Options: Auditd::LogPriority
+  * Default:
+
+##### `facility`:
+
+  'LOG_INFO'
+  * Valid Options: Auditd::LogFacility
```

Comment:
Dunno, was something that came up in our discussion and never reached a good conclusion. It's kind of slippery to make expanding them standard.